report/src/main/webapp/app/services/history.service.js

```diff
@@ -0,0 +1,136 @@
+/*
+  * AET
+  *
+  * Copyright (C) 2013 Cognifide Limited
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License");
+  * you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  *      http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+define(['angularAMD', 'endpointConfiguration'], function (angularAMD) {
+  'use strict';
+  angularAMD.factory('historyService', historyService);
+
+  /**
+   * Service responsible for fetching suite's history
+   */
+  function historyService($rootScope, $http, endpointConfiguration) {
+    var suiteHeaders;
+    $rootScope.endpointUrl = endpointConfiguration.getEndpoint().getUrl;
+    var service = {
+      getNextVersion: getNextVersion,
+      getPreviousVersion: getPreviousVersion,
+      fetchHistory: fetchHistory,
+    };
+
+    return service;
+
+    function fetchHistory(currentVersion, fetchCallback) {
+      $rootScope.selectedVersion = currentVersion;
+      getSuiteHistory(suiteHeaders, $rootScope, $http, function () {
+        fetchCallback();
+      });
+    }
+
+    function buildApiPath($allParametersList) {
+      return $rootScope.endpointUrl + 'history' + '?' +
+        $allParametersList[0] + '&' +
+        $allParametersList[1] + '&' +
+        $allParametersList[2];
+    }
+
+    function getPreviousVersion(currentVersion) {
+      var prevVersion = null;
+      $rootScope.suiteHeaders.forEach(function(suiteHeader, index) {
+        if(suiteHeader.version === currentVersion) {
+          if($rootScope.suiteHeaders[index + 1]) {
+            prevVersion =  $rootScope.suiteHeaders[index + 1].version;
+          }
+        }
+      });
+      return prevVersion;
+    }
+
+    function getNextVersion(currentVersion) {
+      var nextVersion = null;
+      $rootScope.suiteHeaders.forEach(function(suiteHeader, index) {
+        if(suiteHeader.version === currentVersion) {
+          if($rootScope.suiteHeaders[index - 1]) {
+            nextVersion =  $rootScope.suiteHeaders[index - 1].version;
+          }
+        }
+      });
+      return nextVersion;
+    }
+
+    function getSuiteHistory(suiteHeaders, $rootScope, $http, historyCallback) {
+      $rootScope.data = 'test';
+      var cUrl = new URL(window.location.href);
+      var company = cUrl.searchParams.get('company');
+      var project = cUrl.searchParams.get('project');
+      var suite = cUrl.searchParams.get('suite');
+      if (suite === null) {
+        var correlationId = cUrl.searchParams.get('correlationId');
+        suite = correlationId.split('-')[2];
```

Comment:
It is possible that `company` or `project` or `suite` values contain `-` character - then this line will not work correctly.
I think it can be done like this:
```
var cp = company + '-' + project + '-';
var suiteNameWithTimestamp = correlationId.substr(a.indexOf(cp) + cp.length)
// e.g. now: suiteNameWithTimestamp == 'suite-name-123123123` - we need to cut timestamp from here
```